src/Redux/missions/MissionsSlice.js

```diff
@@ -0,0 +1,77 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/missions';
+
+export const fetchMissions = createAsyncThunk(
+  'missions/fetchMissions',
+  async () => {
+    try {
+      const response = await fetch(url);
+      return response.json() || [];
+    } catch (error) {
+      return 'There was an error and no missions were returned';
+    }
+  },
+);
+
+const initialState = {
+  missions: [],
+  status: 'idle',
+  reserved: false,
+};
+
+const data = (missions) => {
+  const missionsList = [];
+  missions.forEach((mission) => {
+    missionsList.push({
+      mission_id: mission.mission_id,
+      mission_name: mission.mission_name,
+      description: mission.description,
+    });
+    return missionsList;
+  });
+  return missionsList;
```

Comment:
- [ ] Great job with your team's project but I think you need to check how you update your store to ensure things are not buggy. If a mission or a rocket is booked, once the browser refreshes, the update is gone and back to default -as shown below:
 
![Studio_Project_V1](https://github.com/AndreaM2429/Space_Travelers_Hub/assets/99715640/0ba61d38-2408-46be-968c-b42e0e7bb7ca)

  - Kindly find a way to fix this anomaly on both your mission & rockets

Reply:
Hello @AbayomiOlaoye I appreciate your time to relieve the project but I want to clarify that the requirements said that we don't have to preserve the data on the browser as you mentioned.
[Here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md) is the link to the requirements.

Reply:
@AbayomiOlaoye I have to agree with @AndreaM2429. This change is not a requirement. What is important is the fact that if you navigate between the pages, the state is maintained. However, in order to have that maintained state when the window reloads, the data should have persisted somewhere like in a database or local storage with that given change to be maintained.